python/ray/tune/integration/docker.py

```diff
@@ -0,0 +1,177 @@
+import os
+import subprocess
+
+import yaml
+from ray import services
+from ray.autoscaler._private.command_runner import DockerCommandRunner
+from ray.autoscaler.tags import NODE_KIND_HEAD, NODE_KIND_UNMANAGED, \
+    NODE_KIND_WORKER, TAG_RAY_NODE_KIND
+from ray.tune.syncer import NodeSyncer
+from ray.tune.sync_client import SyncClient
+
+
+class _WrappedProvider:
+    def __init__(self, config_path):
+        from ray.autoscaler._private.util import validate_config
+        from ray.autoscaler.node_provider import _get_node_provider
```

Comment:
I think if you rebase onto master this will not work - these are now private methods?

Reply:
In the refactoring this got introduced into the `KubernetesSyncer`, too. Main question is what the alternative would be - reimplementing seems like a waste? Are there public alternatives to these? (I can look into that tomorrow)

Reply:
I think it's ok to keep as private method and then just raise a concern that we need them (or ideally, we can abstract out a part of the commandrunner?)